src/main/java/uk/gov/companieshouse/company/profile/converter/CompanyProfileConverter.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.companieshouse.company.profile.converter;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.bson.Document;
+import org.bson.json.JsonWriterSettings;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.company.Data;
+
+@ReadingConverter
+@Component
+public class CompanyProfileConverter implements Converter<Document, Data> {
+
+    private final ObjectMapper mapper;
+
+    public CompanyProfileConverter(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public Data convert(Document source) {
+        // Exclude properties with null values from being serialised
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```

Comment:
Probably better to create new ObjectMapper object for this instead of using default one, same will be useful for WritingConverter as well

Reply:
Created a custom object mapper with a qualifier to be used in mongo reading/writing converter, while preserving the default spring one (marked as primary).